### PR TITLE
Todo

### PR DESCRIPTION
**Pull request by 1v4n4**
Do It is an application that helps users to organize their lives. They can create (edit and delete) project with titles and optional descriptions and for each project an unlimited amount of to-dos. 
- Tasks must be named, they have priority and status tags, as well as deadlines.
- Status can be changed in the task's table, the rest of its content can be edited or whole tasks can be deleted. 
- If the deadline is not selected, it is due today by default. 
- If priority is not selected, it is lowest by default. 
- Status is unfinished by default. 
- Users can add comments if they want.
- If the project is deleted, all its tasks go along. If the project is renamed, tasks are reassigned.
- Users can see today's unfinished tasks in a separate section.
- Projects must have different names but tasks don't.

Default projects and tasks, can't be changed, despairing when the user adds a new one.

There are two selections of tasks:
- When users select a project(by clicking on the arrow above the project's name) middle section is opened with tasks that belong to that project.
- In the far-right sections are today's unfinished tasks from all projects.

In two modules some linters are disabled, by using inline code. Two modules are completely disabling-free.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
+import { setDate } from './modules/helpers';
+// eslint-disable-next-line
```
Comment (addod19) [inline src/index.js]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+import {
+  Project, projectsToDom, deleteProject, checkProject,
+} from './modules/projects';
+// eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:5]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+const addForm = document.getElementById('addForm');
+const getName = document.getElementById('nameInput');
+const getDescription = document.getElementById('descriptionInput');
+const first = document.getElementById('first');
+const projectsContainer = document.getElementById('projectsContainer');
+const tableArticle = document.getElementById('tableArticle');
+const editForm = document.getElementById('editForm');
+const editTaskForm = document.getElementById('editTaskForm');
+
+// eslint-disable-next-line
```
Comment (addod19) [inline src/index.js]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  myProjects = JSON.parse(window.localStorage.getItem('myProjects'));
+}
+
+function setProjects() {
+  window.localStorage.setItem('myProjects', JSON.stringify(myProjects));
+}
+
+let currentProject = '';
+
+// eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:65]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+  myTasks = JSON.parse(window.localStorage.getItem('myTasks'));
+}
+
+function setTasks() {
+  console.log('in setTasks');
+  window.localStorage.setItem('myTasks', JSON.stringify(myTasks));
+}
+
+document.addEventListener('DOMContentLoaded', projectsToDom(), makeToday());
+// eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:77]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    let confirm = true;
+
+    if (!newTitle) {
+      alert('Title must not be empty');
+      confirm = false;
+      return;
+    }
+
+    if (confirm === false) {
+      // eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:222]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+    makeTaskSection(currentProject, selectedTasks);
+    makeToday();
+    taskForm.reset();
+  }
+});
+
+tableArticle.addEventListener('click', (e) => {
+  e.preventDefault();
+  const clicked = e.target;
+  // eslint-disable-next-line
```
Comment (addod19) [inline src/index.js]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+    const taskToEdit = myTasks[index];
+    makeEditForm(taskToEdit);
+    editTaskForm.addEventListener('click', (e) => {
+      e.preventDefault();
+      const clicked = e.target;
+      if (clicked.classList.contains('dismissEditTask')) {
+        editTaskForm.innerHTML = '';
+      }
+      if (clicked.classList.contains('confirmEditTask')) {
+        // eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:282]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+        let confirm = true;
+
+        if (!editTitle) {
+          alert('Title must not be empty');
+          confirm = false;
+          return;
+        }
+
+        if (confirm === false) {
+          // eslint-disable-next-line
```
Comment (addod19) [inline src/index.js:297]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/helpers.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  if (dd < 10) {
+    dd = `0${dd}`;
+  }
+  if (mm < 10) {
+    mm = `0${mm}`;
+  }
+  today = `${yyyy}-${mm}-${dd}`;
+  return today;
+};
+// eslint-disable-next-line
```
Comment (addod19) [inline src/modules/helpers.js]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/projects.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line
```
Comment (addod19) [inline src/modules/projects.js]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/tasks.js [hunk 12]
```diff
@@ -0,0 +1,211 @@
+// eslint-disable-next-line
```
Comment (addod19) [inline src/modules/tasks.js:1]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/tasks.js [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+const first = document.getElementById('first');
+const taskForm = document.getElementById('taskForm');
+const editTaskForm = document.getElementById('editTaskForm');
+const tableArticle = document.getElementById('tableArticle');
+const todayContainer = document.getElementById('todayContainer');
+
+let countProject = false;
+// eslint-disable-next-line
```
Comment (addod19) [inline src/modules/tasks.js:24]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/tasks.js [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+
+  first.appendChild(tasksTitleDiv);
+  tasksToDom(selected);
+  countProject = true;
+};
+
+const makeTaskForm = () => {
+  // const taskForm = document.createElement('form');
+  // taskForm.className = "taskForm border border-dark px-2 mx-5 ";
+  // taskForm.setAttribute('id', 'taskForm');
```
Comment (addod19) [inline src/modules/tasks.js]: Kindly remove the commented piece of code, use "git history" instead

File: src/modules/tasks.js [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+    document.getElementById('datePicker').setAttribute('min', setDate());
+    countForm = true;
+  } else {
+    taskForm.innerHTML = '';
+    countForm = false;
+  }
+};
+
+const changeStatus = (index) => {
+  // eslint-disable-next-line
```
Comment (addod19) [inline src/modules/tasks.js:162]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/tasks.js [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+      <button type='submit' class='confirmEditTask btn btn-secondary' id='confirmEditTask'>Edit task</button>
+      <button type='button' class='dismissEditTask btn btn-light m-3' id='dismissEditTask'>Dismiss</button>
+    `;
+
+  document.getElementById('datePicker1').setAttribute('min', setDate());
+};
+
+const makeToday = () => {
+  todayContainer.innerHTML = '';
+  // eslint-disable-next-line
```
Comment (addod19) [inline src/modules/tasks.js:212]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

File: src/modules/tasks.js [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+              <li class="list-group-item bg-light">Status: ${modalStatus(task.finished)}</li>
+              <li class="list-group-item bg-light">Deadline: ${task.deadline}</li>
+            </ul>
+
+          </div>`;
+  });
+};
+
+export {
+  // eslint-disable-next-line
```
Comment (addod19) [inline src/modules/tasks.js:233]: Kindly avoid the ESLInt inline disabling rule, it is the same as altering the ESLint config files

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @1v4n4, great job on this project :+1: The README is professional :100: Commit messages easy to understand :slightly_smiling_face: PR very descriptive :smile: FIles are separated in modules :+1: Local storage implemented to persist data :1st_place_medal: 
Let us address the comment below in order to get approval

- There should be a default project, currently we do not have one. Let us kindly add a default project
- Let us improve the functionality, for example; when we have 2 projects ie. project1 and project2, when I click on the project1 I should be able to see all the tasks in project1 and when I click on the project2 I should be able to see all the tasks in the project2, but in our case, when I click on project1 I see all tasks in both project1 and project2 which is not how the app should function. Let us check it and fix it
- Do not make any changes to the linter files
- Kindly remove all commented pieces of code


Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/index.js (hunk 1 above)
Comment (1v4n4) [reply src/index.js]: Actualy, this is what i meant by  disabling the linters. I didn't changed the files.

File: src/modules/tasks.js (hunk 14 above)
Comment (1v4n4) [reply src/modules/tasks.js]: Done!

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED 🔴 🛑 
-----------
Hello @1v4n4, Great job yp¡ou are doing in this project and good job working on the required changes 👏 
However, there is minor changes to make to complete this project 🛠️ see them below 👀 👇 
---------

- [ ] When I edit a  project I got the message `You haven't change anything` when I change it. the project change but I got this message. Please fix this. 
- [ ]  There is a bug in the edit task when I edit the second task it changes the first one. Please fix this. 
- [ ] When I try to edit any task the edit doesn't work y keeps the same information.
- [ ] If I delete the task the edit form keeps showing and if I click on the button form the task s I delete come back.  
------------- 
Please make these required changes and send it again for another review 💪 
HAPPY CODING!!!

programming-punching-keyboard-coding-anger-13716030304

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] Git flow is used properly.
- [ ] README includes instructions to run the project locally.
- [ ] App runs perfectly.
## To Improve 🙏
- [x] Please do not disable eslint rules in every module, but fix the errors detected by them instead.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by GzimAsani — APPROVED** [review]
# STATUS: Project Accepted :green_circle: 

# Hello @1v4n4, thank you for implementing all the required changes. Good luck with your next projects :+1: 

# HAPPY CODING :computer: :desktop_computer:

**Pull request merged**